### PR TITLE
[r] SOMASparseNDArray: sparseMatrix zero-based shim to facilitate soma_joinid lookup

### PR DESCRIPTION
**Pull request by mlin**
Extended discussion: #1232

To load a one-based `Matrix::sparseMatrix` with the contents of a zero-based `SOMASparseNDArray`, we add one to the offsets on each dimension. However, these dimensions are usually populated with `soma_joinid` intended to match the `soma_joinid` column in the `obs` & `var` data frames, and shifting them by one makes this join operation very error-prone. 

Here we introduce a minimal shim for `sparseMatrix` that provides matrix access with zero-based indexes. To make this explicit for the user, we rename `SOMASparseNDArray$read_sparse_matrix()` to `SOMASparseNDArray$read_sparse_matrix_zero_based()`. The shim supports only minimal access operations, which is intentional to prevent "mixing" it with conventional one-based objects. If needed, the fully-featured `sparseMatrix` is recovered by calling `as.one.based()` on the shim. Thus, the default behavior is to match `soma_joinid` but an advanced user can explicitly change to one-based indexing for further use in R.

This is a refinement of the first attempt #1306 which was far more complex and error-prone, attempting a subclass with selected methods overridden instead of the distinct wrapper shim shown here.

### CODE REVIEW

**Comment by mlin** [comment]
@eddelbuettel Updated NAMESPACE and added the Rd's for the new methods. Perhaps we leave updating the existing Rd's to a followup since, as you noted, it's a lot of unrelated changes.

File: apis/r/NAMESPACE [hunk 1]
```diff
@@ -1,8 +1,12 @@
 # Generated by roxygen2: do not edit by hand
 
+S3method("[",matrixZeroBasedView)
+S3method("[<-",matrixZeroBasedView)
```
Comment (eddelbuettel) [inline apis/r/NAMESPACE:4]: Thanks for adding it.

**Review by eddelbuettel — APPROVED** [review]
I think this is good to go.

**Comment by mlin** [comment]
@eddelbuettel thanks!

@mojaveazure @aaronwolen double-checking just since this was such a tricky topic -- any last comments/concerns?

**Pull request merged**